core/src/main/java/org/hibernate/ogm/storedprocedure/ProcedureQueryParameters.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.storedprocedure;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Named or positional parameters to pass to a stored procedure.
+ *
+ * @author Davide D'Alto
+ */
+public class ProcedureQueryParameters {
```

Comment:
Why did You not use org.hibernate.ogm.dialect.query.spi.QueryParameters ? What is reason for create new class that looks like org.hibernate.ogm.dialect.query.spi.QueryParameters?

Reply:
I didn't want to deal with the TypeGridValue, I'm not sure if it make sense when it comes with stored procedures.

Reply:
Hm ... What is problem with TypeGridValue ?

Reply:
Add complexity that I'm not sure we need

Reply:
Okey. May be ... you are right. Anyway, we will have possibility return to using TypeGridValue .

Reply:
Yes, as I said. I'm not completely sure about this solution. But I prefer to start with something simple and add the missing bit later

Reply:
I see that method "executeBackendQuery" uses org.hibernate.ogm.dialect.query.spi.QueryParameters . I think it is not logical. Queries use one class for parameters but stored procedures use other class for parameters. What do you think about it?

Reply:
Why not? calling a query is different from calling a stored procedure, in fact we use two different methods to execute them. What's so surprising if they also use 2 different classes for the parameters? By the way, parameters for stored procedures are a a bit different than parameters for queries, because you also have to consider if they are IN, OUT, etc.

Reply:
okey.  Time will tell .